src/controllers/confirm.company.controller.ts

```diff
@@ -17,10 +18,36 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+    if (isStatusEligible(company)) {
+      create(req);
+      const url = CONFIRMATION_STATEMENT_TRADING_STATUS.replace(":companyNumber", company.companyNumber);
+      return res.redirect(url);
+    } else {
+      return res.render(Templates.INVALID_STATUS);
+    }
+  } catch (e) {
+    next(e);
+  }
+};
+
+const isStatusEligible = (company: CompanyProfile): boolean => {
```

Comment:
move the eligible check out of the controller into it's own file. There is a validators folder now so makes sense to have a company status validator in there. Although check with Ian about wether the checks should be done in the api.

Reply:
Flagged ticket pending discussion.